### PR TITLE
port daemonp

### PR DESCRIPTION
**Pull request by brotzeit**

### CODE REVIEW

File: rust_src/src/emacs.rs [hunk 1]
```diff
@@ -34,4 +38,21 @@ pub fn invocation_directory() -> LispObject {
     unsafe { Fcopy_sequence(globals.Vinvocation_directory) }
 }
 
+/// Return non-nil if the current emacs process is a daemon.
+/// If the daemon was given a name argument, return that name.
+#[lisp_fn]
+pub fn daemonp() -> LispObject {
+    unsafe {
+        if is_daemon() {
+            if !daemon_name.is_null() {
```
Comment (shaleh) [inline rust_src/src/emacs.rs]: Flipping the check so it does not need to be negated I think will read better.

    if daemon_name.is_null() {
        Qt
    } else {
        build_string(daemon_name)
    }

Comment (daichimukai) [reply rust_src/src/emacs.rs]: How about avoiding the nesting?

```rust
if is_daemon() && !daemon_name.is_null() {
    return build_string(daemon_name);
}

is_daemon().into()
```

Comment (shaleh) [reply rust_src/src/emacs.rs]: The only negative there is the double call to `is_daemon`.

**Comment by shaleh** [comment]
My understanding is we cannot inline anything currently. If you specify it the compiler will ignore you. All of the Rust code is compiled into a library which prevents the inline analysis. We used to specify inline attributes and stopped.

**Comment by daichimukai** [comment]
@shaleh Interesting. Thanks for information.

**Pull request merged**